File: VB/GenerateUserFriendlyId.Module/Module.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.ExpressApp
Imports GenerateUserFriendlyId.Module.BO
Imports GenerateUserFriendlyId.Module.Utils

Namespace GenerateUserFriendlyId.Module
	Public NotInheritable Partial Class GenerateUserFriendlyIdModule
		Inherits ModuleBase
		Public Sub New()
			InitializeComponent()
		End Sub
		Public Overrides Overloads Sub Setup(ByVal application As XafApplication)
			MyBase.Setup(application)
			If (Not XafTypesInfo.IsInitialized) Then
				XafTypesInfo.Instance.RegisterEntity("Document", GetType(IDocument), GetType(BasePersistentObject))
				SequenceGeneratorInitializer.Register(application)
			End If
		End Sub
	End Class
End Namespace

File: CS/GenerateUserFriendlyId.Module/Module.cs
using DevExpress.ExpressApp;
using GenerateUserFriendlyId.Module.BO;
using GenerateUserFriendlyId.Module.Utils;

namespace GenerateUserFriendlyId.Module {
    public sealed partial class GenerateUserFriendlyIdModule : ModuleBase {
        public GenerateUserFriendlyIdModule() {
            InitializeComponent();
        }
        public override void Setup(XafApplication application) {
            base.Setup(application);
            XafTypesInfo.Instance.RegisterEntity("Document", typeof(IDocument), typeof(BasePersistentObject));
            SequenceGeneratorInitializer.Register(application);
        }
    }
}